### DISCUSSION TITLE
How could I center the caption of an image?

### DISCUSSION THREAD

**Discussion by hardianlawi** [mmistakes/minimal-mistakes#2682] [Q&A] [answered]
Hi, I am a noobie in both html and CSS. This [page](https://mmistakes.github.io/minimal-mistakes/docs/helpers/) shows how to add an image with caption, but I would like to position the caption in the middle of the image. What is the cleanest way to do that? A follow-up question, how could I create a reference of the image?

**Comment by iBug** [comment, marked as answer]
To center the image caption, the canonical way is via CSS:

```scss
figcaption {
  margin-left: auto;
  margin-right: auto;
}
```

For "reference", can you elaborate on what it exactly is?

**Comment by hardianlawi** [comment]
@iBug Thanks for your prompt reply. Where should I add that CSS? Is it in `_includes/figures`? Currently it is

```

<figure class="{{ include.class }}">
  <img src="{{ include.image_path | relative_url }}"
       alt="{% if include.alt %}{{ include.alt }}{% endif %}">
  {% if include.caption %}
    <figcaption>
      {{ include.caption | markdownify | remove: "<p>" | remove: "</p>" }}
    </figcaption>
  {% endif %}
</figure>

```

I tried this and it didn't work.
```
<html>
<head>
<style>

figcaption {
  margin-left: auto;
  margin-right: auto;
}

</style>
</head>
<body>

<figure class="{{ include.class }}">
  <img src="{{ include.image_path | relative_url }}"
       alt="{% if include.alt %}{{ include.alt }}{% endif %}">
  {% if include.caption %}
    <figcaption>
      {{ include.caption | markdownify | remove: "<p>" | remove: "</p>" }}
    </figcaption>
  {% endif %}
</figure>

</body>
</html>
```

> For "reference", can you elaborate on what it exactly is?

I am thinking of a link to the image which I could include in a paragraph, e.g.

```
[Image.jpg]


From the [above](point to Image.jpg) image, ....
```

**Comment by iBug** [comment]
Please refer to the documentation about customizing the theme's CSS:

<https://mmistakes.github.io/minimal-mistakes/docs/stylesheets/#customizing>

For the "image reference" part, you can add an `id` attribute to the image and then use `[link text](#your-id)` to refer to the annotated image.

**Comment by hardianlawi** [comment]
Got it. I will check the documentation on that. Thanks so much for the help! Will close this issue for now.
